### PR TITLE
Enable HTTP/2 for internal communication by default

### PR DESCRIPTION
**Pull request by wendigo**
This is it :)

### CODE REVIEW

**Review by martint — APPROVED** [review]

**Comment by wendigo** [comment]
@electrum ptal

**Comment by mosabua** [comment]
I assume @wendigo and @electrum are still running with this.

**Comment by wendigo** [comment]
@mosabua yes sir

File: core/trino-main/src/main/java/io/trino/server/DisconnectionAwareAsyncResponse.java [hunk 1]
```diff
@@ -83,55 +78,81 @@ public void onStartAsync(AsyncEvent event) {}
...
     }
 
     @Override
     public boolean resume(Object response)
     {
-        if (clientDisconnected.get()) {
+        try {
+            return delegate.resume(response);
+        }
+        catch (Exception e) {
```
Comment (electrum) [inline core/trino-main/src/main/java/io/trino/server/DisconnectionAwareAsyncResponse.java:94]: `RuntimeException` for all of these

**Review by electrum — APPROVED** [review]

**Comment by wendigo** [comment]
Screenshot 2024-09-23 at 22 09 58

orange is http/2, looks good

**Pull request closed without merging**